### DISCUSSION TITLE
Colors appear washed out and darker background

### DISCUSSION THREAD

**Discussion by adelejjeh** [arcticicestudio/nord#226] [Help & Support]
![Screen Shot 2021-02-10 at 8 34 11 AM](https://user-images.githubusercontent.com/14184522/107524159-1c9af980-6b7b-11eb-8405-bed2c4cc7096.png)

**Comment by arcticicestudio** [comment]
Hi @adelejjeh :wave:, that's pretty little information 😉 

In order to ensure the problem is related to the Nord theme you need to reproduce it with a minimal and clean configuration. This can be done by creating new configuration file with any name (e.g. `testrc`) and run Vim with the following command: `vim -u testrc`
For me as a theme author it is not possible to know if the theme is the root cause of the problem when there are other attributes set in the configuration because each one might break the theme.

Here's an minimal test configuration you can use to try to reproduce the problem (supposed you're using [vim-plug][]):

```vim
call plug#begin(expand('~/.vim/plugged'))
Plug 'arcticicestudio/nord-vim'
call plug#end()

set nocompatible
if (has("termguicolors"))
  set termguicolors
endif

syntax enable
colorscheme nord
```



As you can see in the example configuration the `termguicolors` setting is enabled because the [`*_term` attributes][gh-blob-term_attr] make use of the colors that are provided by the terminal color scheme which might be for example [iTerm2][ref-iterm] or [Terminal.app][ref-term.app] in your case. The [`*_gui` variables][gh-blob-gui_attr] on the other side are used when Vim runs in GUI mode. So in order to run Vim in a terminal but use colors like in GUI mode the `termguicolors` variable must be set and your terminal emulator must support “true colors“ (24bit).
→ Did you install the corresponding Nord theme for your terminal, e.g. [Nord iTerm2][n-iterm2]?

Unfortunately terminal colors are quite complex due to their long history, but modern terminal emulators should be all capable of rendering "True Colors". You should make sure to check out the great [“Terminal Colors“ gist][gist-tc] that provides many details about color rendering in terminal emulators, tracking of supported terminal emulator projects and answers to common confusions about terminal colors. The [“Colours in terminal“ gist][gist-cit] is also quite helpful to learn more about the actual rendering engines and standards used by these terminal emulators.
There are also a lot of blog posts like [this][bp1] and [this][bp2] one that help to handle different use case with various terminal emulators in combination with different CLI and TUI applications.

→ Could you please try to either use the example configuration or set the `termguicolors` setting in your current configuration and check whether colors are then displayed correctly?

[ref-term.app]: https://github.com/arcticicestudio/nord-terminal-app/blob/master/src/xml/Nord.terminal#L23-L31
[ref-iterm]: https://github.com/arcticicestudio/nord-iterm2/blob/master/src/xml/Nord.itermcolors#L187-L199
[n-iterm2]: https://github.com/arcticicestudio/nord-iterm2
[vim-plug]: https://github.com/junegunn/vim-plug
[bp1]: https://sts10.github.io/2016/06/15/true-hex-colors-in-vim-with-iterm2.html
[bp2]: https://tomlankhorst.nl/iterm-tmux-vim-true-color
[gist-tc]: https://gist.github.com/XVilka/8346728
[gist-cit]: https://gist.github.com/sindresorhus/bed863fb8bedf023b833c88c322e44f9
[gh-blob-term_attr]: https://github.com/arcticicestudio/nord-vim/blob/ea7ff9c3/colors/nord.vim#L37-L49
[gh-blob-gui_attr]: https://github.com/arcticicestudio/nord-vim/blob/ea7ff9c3/colors/nord.vim#L19-L35

**Comment by darthmall** [comment]
I'm having a similar problem and I think I've isolated the issue to tmux. When I run NeoVim with the test configuration above outside of tmux (I'm using Alacritty v0.7.2), the colors are correct. When I start tmux and then open NeoVim with the same test config, the colors look similar to the ones in the original poster's screenshot.

I'm on Ubuntu 18.04, so the version of tmux is 2.6, which is from 2017. I _think_ I have this working on my other computer which is a much more recent release of Ubuntu (pretty sure it's 20.10), so I'll double-check that it's working tonight and report back with the version of tmux I have. Could be that you just need to update tmux.

**Comment by darthmall** [comment]
I don't have the same problem with tmux 3.1b, so it's conceivable that this is a tmux bug that has been fixed in more recent versions.

**Comment by chengleqi** [comment]
Same!
I put the following sentences in the `~/.tmux.conf` file and it solved.
```
set -g default-terminal screen-256color
set-option -ga terminal-overrides ",*256col*:tc"
```

**Comment by vendion** [comment]
I'm having a similar issue, although I can't seem to figure out the cause.

![Screenshot_20220622_103004](https://user-images.githubusercontent.com/145018/175042250-1ec7f2c5-2607-4297-b858-2c91ca5433b9.png)

As far as my Vim config goes my color here are the color and color scheme settings I used

```
" Color Scheme stuff
if has('termguicolors')
let &t_8f = "\<Esc>[38;2;%lu;%lu;%lum"
let &t_8b = "\<Esc>[48;2;%lu;%lu;%lum"
set t_Co=256
set termguicolors
endif

colorscheme nord
augroup nord-theme-overrides
let g:nord_italic = 1
let g:nord_italic_comments = 1
let g:nord_underline = 1
let g:nord_uniform_diff_background = 1
autocmd!
" Use 'nord6' as foreground color for all comments.
autocmd ColorScheme nord highlight Comment ctermfg=242 guifg=#ECEFF4
autocmd ColorScheme nord highlight LineNr ctermfg=242 guifg=#ECEFF4
augroup END
```
Note that without these lines, I have no color at all inside of tmux

```
let &t_8f = "\<Esc>[38;2;%lu;%lu;%lum"
let &t_8b = "\<Esc>[48;2;%lu;%lu;%lum"
set t_Co=256
```

I did try setting the tmux settings suggested by @chengleqi without any luck.

I'm running Arch Linux, with tmux 3.3_a, and vim 8.2.5046

**Comment by chengleqi** [comment]
> 

I have this problem again now. lol.
I'm running Ubuntu20.04, with tmux 3.0a and nvim v0.6.1

**Comment by vendion** [comment]
It seems that using the following in _$HOME/.tmux.conf_ mostly fixes it

```
set -g default-terminal "screen-256color"
set-option -ga terminal-overrides ",xterm-256color*:Tc"
```

![Screenshot_20220623_111035](https://user-images.githubusercontent.com/145018/175320180-b17b583a-ada4-439a-8d0b-0ae370a18bab.png)

I should also have added in my other comment my terminal emulator is Konsole if that helps.
Now it seems it is just the odd background color in the comments.

**Comment by chengleqi** [comment]
> 

it's weird. It worked on wezterm, but not on my alacritty. Maybe I should switch to Wezterm?😂

**Comment by lafriakh** [comment]
> it's weird. It worked on wezterm, but not on my alacritty. Maybe I should switch to Wezterm?😂

You can fix it by setting the term property in Alacritty config to:
```
env:
  TERM: xterm-256color
```

**Comment by chengleqi** [comment]
> > it's weird. It worked on wezterm, but not on my alacritty. Maybe I should switch to Wezterm?joy
> 
> You can fix it by setting the term property in Alacritty config to:
> 
> ```
> env:
>   TERM: xterm-256color
> ```

It Worked! thanks so much.

**Comment by vendion** [comment]
> > it's weird. It worked on wezterm, but not on my alacritty. Maybe I should switch to Wezterm?joy
> 
> You can fix it by setting the term property in Alacritty config to:
> 
> ```
> env:
>   TERM: xterm-256color
> ```

Setting that in _$HOME/.tmux.conf_ also seems to fix it using Konsole + Tmux.

**Comment by svengreb** [comment]
Thank you for your patience! 🙏🏼
It‘s been a while [since I had free time to focus more on Nord, and my open source projects in general](https://github.com/orgs/nordtheme/discussions/183#user-content-todays-scope), and invest time in this issue due to [work-life balance](https://github.com/orgs/nordtheme/discussions/183#work-life-balance).

I recently published the first [“Northern Post — The state and roadmap of Nord“ announcement](https://github.com/orgs/nordtheme/discussions/183) which includes all details about the plans and future of the _Nord_ project, including [the goal of catching up with the backlog](https://github.com/orgs/nordtheme/discussions/183#catching-up-on-the-backlog). This issue is part of the backlog and therefore I want to triage and process it to get one step closer to a “clean state“. Read the [announcement about reaching the “clean“ contribution triage state](https://github.com/orgs/nordtheme/discussions/202) in Nord‘s discussions for more details about the goal.

Due to the scope of this issue of not being “purely technical“, or being related to the project‘s code at all, I‘ll convert it into a discussion for [Nord‘s central community hub on GitHub](https://github.com/orgs/nordtheme/discussions). This way the repository is easier to maintain by separating conversations about the actual code while also making help requests, ideas and other contributions from users more visible and supportable by the community.
